### PR TITLE
Generate cfg_condition on sys mode

### LINKED ISSUE #645: sys: hide behind a flag certain features
in sys generation it would be useful to mark certain ffi functions behind a named feature flag

### PR DESCRIPTION
**Pull request by ceyusa**
fixes #645

### CODE REVIEW

File: src/codegen/sys/functions.rs [hunk 1]
```diff
@@ -155,6 +156,18 @@ pub fn generate_other_funcs(
...
 
+fn generate_cfg_configure(
+    w: &mut Write,
+    functions: &Vec<&Function>,
+    commented: bool,
+)-> Result<()> {
+    let with_cfg_condition = functions.iter().find(|f| f.cfg_condition.is_some());
+    if let Some(info) = with_cfg_condition {
+        cfg_condition(w,  &info.cfg_condition, commented, 1)?;
+    }
```
Comment (EPashkin) [inline src/codegen/sys/functions.rs]: Please use something like next here:
```
    let cfg_condition = configured_functions
        .iter()
        .filter_map(|f| f.cfg_condition.clone())
        .next();
    cfg_condition(w,  &cfg_condition, commented, 1)?;
```

Comment (EPashkin) [reply src/codegen/sys/functions.rs]: Maybe without `clone`

**Comment by ceyusa** [comment]
> Should probably also exist for types (e.g. `GstEGLDisplay`)

Do structs and traits handle #[cfg(...)] ???

File: src/codegen/sys/functions.rs (hunk 1 above)
Comment (ceyusa) [reply src/codegen/sys/functions.rs]: without clone(): error[E0507]: cannot move out of borrowed content

Comment (EPashkin) [reply src/codegen/sys/functions.rs]: Hm. `clone` can't be removed, so better leave as is, sorry.

**Comment by sdroege** [comment]
> Do structs and traits handle #[cfg(...)] ???

Yes, every item (functions, type declarations, code blocks, etc)

**Comment by EPashkin** [comment]
Please update README.md in "The FFI mode .." section too
Thanks in advance.

File: README.md [hunk 2]
```diff
@@ -67,6 +67,19 @@ status = "generate"
...
+[[object]]
+name = "GstGL.GLDisplayEGL"
+status = "generate"
+cfg_condition = "feature = \"egl\""
+    [[object.function]]
+    pattern = ".*"
+    cfg_condition = "feature = \"egl\""
+```
+
+
```
Comment (EPashkin) [inline README.md]: Please remove one empty line,

**Comment by EPashkin** [comment]
@ceyusa Big thanks, please fix "update README.md with cfg_conditions for sys mode " and I will merge PR.

**Comment by ceyusa** [comment]
> @ceyusa Big thanks, please fix "update README.md with cfg_conditions for sys mode " and I will merge PR.

Done :smiley:

**Pull request merged**